util/logtools/encounter_tools.ts

```diff
@@ -286,3 +286,191 @@ export class EncounterCollector extends EncounterFinder {
     this.lastSeal = undefined;
   }
 }
+
+class TLUtilFunctions {
+  timeFromDate(date: Date | undefined): string {
+    if (date)
+      return date.toISOString().slice(11, 19);
+    return 'Unknown_Time';
+  }
+
+  dayFromDate(date: Date | undefined): string {
+    if (date)
+      return date.toISOString().slice(0, 10);
+    return 'Unknown_Date';
+  }
+
+  durationFromDates(start: Date | undefined, end: Date | undefined): string {
+    if (!(start && end))
+      return 'Unknown_Duration';
+    const ms = end.valueOf() - start.valueOf();
+    const totalSeconds = Math.round(ms / 1000);
+    const totalMinutes = Math.floor(totalSeconds / 60);
+    if (totalMinutes > 0)
+      return totalMinutes.toString() + 'm';
+    return (totalSeconds % 60).toString() + 's';
+  }
+
+  toProperCase(str: string): string {
+    return str.split(' ').map((str) => {
+      const cap = str[0]?.toUpperCase();
+      const lower = str.slice(1);
+      if (cap)
+        return `${cap} ${lower}`;
+    }).join(' ');
+  }
+
+  leftExtendStr(str: string | undefined, length: number | undefined): string {
+    if (!str)
```

Comment:
```suggestion
    if (str === undefined)
```

Since empty string is false...